design/dsl/endpoint_test.go

```diff
@@ -148,9 +146,17 @@ func assertEndpointMetaData(t *testing.T, actual design.MetadataExpr, expected d
 }
 
 func assertEndpointRequestResponse(t *testing.T, assertType string, actual design.DataType, expected *design.UserTypeExpr) {
```

Comment:
It seems this could be generalized to handle all required user type comparisons by changing the signature to something like:
```go
func assertUserType(t *testing.T, context string, actual design.DataType, expected *design.UserTypeExpr)
```
Where `context` contains a generic string used to prefix error rmessages:
```go
t.Errorf("expected %s to be a user type, got %s", context, actual.Name())
```